src/main/java/stsjorbsmod/actions/ConsumeCardIfFatalAction.java

```diff
@@ -0,0 +1,35 @@
+//
+// Source code recreated from a .class file by IntelliJ IDEA
+// (powered by Fernflower decompiler)
+//
+
+package stsjorbsmod.actions;
+
+import com.megacrit.cardcrawl.actions.AbstractGameAction;
+import com.megacrit.cardcrawl.cards.AbstractCard;
+import com.megacrit.cardcrawl.cards.DamageInfo;
+import com.megacrit.cardcrawl.core.AbstractCreature;
+import com.megacrit.cardcrawl.powers.MinionPower;
+
+public class ConsumeCardIfFatalAction extends AbstractGameAction {
+    private DamageInfo info;
+    private AbstractCard card;
+
+    public ConsumeCardIfFatalAction(AbstractCreature target, DamageInfo info, AbstractCard c) {
+        this.info = info;
+        this.setValues(target, info);
+        this.card = c;
+        this.actionType = ActionType.DAMAGE;
+        this.duration = 0.1F;
+    }
+
+    public void update() {
+        if (this.duration == 0.1F && this.target != null) {
+            this.target.damage(this.info);
```

Comment:
In general, any action that's directly invoking `monster.damage` needs a `clearPostCombatActions` check. I think maybe it would be helpful to make a generalized form of the action you implemented for Siphon, something like `DamageWithOnKillEffectAction` that accepts a Runnable; I think there's something like a half dozen CULL cards that could end up using that.

Reply:
A generic do-something-on-kill action would be great, but I'm not sure I know how to do that :o  Could you give me an idea of what that would look like? What do I pass into it?

Reply:
In Java, the usual interface for a generic callback like this is called `Runnable` (if you wanted to use something that took a parameter, it's `Consumer`). Using it would look something like this:

```java
public DamageWithOnKillEffectAction(AbstractCreature target, DamageInfo info, Runnable onKillCallback) {
    // ... assign to fields
}

public void update() {
    // as HealOnFatalDamageAction, except replace the HealAction bit with
    this.onKillCallback.run();
}

// elsewhere, in each individual card...

public void use(/* ... */) {
    Runnable onKillEffect = () -> addToBot(new HealAction(p, p, heal));
    addToBot(new DamageWithOnKillEffectAction(m, new DamageInfo(p, damage), onKillEffect));
}
```